gems/sorbet-runtime/test/types/sealed_module.rb

```diff
@@ -162,4 +162,42 @@ module Opus::Types::Test::SealedModuleSandbox; end
     require_relative './fixtures/sealed_module/whitelist_violation__1.rb'
     require_relative './fixtures/sealed_module/whitelist_violation__2.rb'
   end
+
+  it "returns an empty set if we try to get the subclasses of a sealed-but-never-inherited parent" do
+    parent = Class.new do
+      extend T::Helpers
+      sealed!
+    end
+
+    assert_equal(Set.new, parent.sealed_subclasses)
+  end
```

Comment:
Do you want to add a static test for this case too? You should get a `T.reveal_type` of `T::Set[T.noreturn]`